### PR TITLE
Update PTL examples for v1.6 release

### LINKED ISSUE #138: Review and fix example notebooks
## 📚 Documentation

Reported here: https://github.com/PyTorchLightning/pytorch-lightning/issues/11693#issuecomment-1027770945
[This example](https://pytorch-lightning.readthedocs.io/en/stable/notebooks/lightning_examples/basic-gan.html#C.-GAN) may be more might be using old API and needs to be reviewed and fixed with the latest changes.

Thanks!

cc @borda @rohitgr7

### PR DESCRIPTION
**Pull request by rohitgr7**
# Before submitting

- [x] Was this discussed/approved via a Github issue? (no need for typos and docs improvements)
- [x] Did you make sure to update the docs?
- [x] Did you write any new necessary tests?

## What does this PR do?

Fixes #138

## PR review

Anyone in the community is free to review the PR once the tests have passed.
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

## Did you have fun?

Make sure you had fun coding 🙃

### CODE REVIEW

File: lightning_examples/barlow-twins/barlow_twins.py [hunk 1]
```diff
@@ -23,6 +23,7 @@
 from torchvision.models.resnet import resnet18
 from torchvision.utils import make_grid
 
+AVAIL_GPUS = min(1, torch.cuda.device_count())
```
Comment (kaushikb11) [inline lightning_examples/barlow-twins/barlow_twins.py:26]: Or we could just use `devices="auto"`.

fyi: it's default as well.

File: lightning_examples/basic-gan/gan.py [hunk 2]
```diff
@@ -9,12 +8,12 @@
...
 import torchvision.transforms as transforms
 from pytorch_lightning import LightningDataModule, LightningModule, Trainer
+from pytorch_lightning.callbacks.progress import TQDMProgressBar
 from torch.utils.data import DataLoader, random_split
 from torchvision.datasets import MNIST
 
 PATH_DATASETS = os.environ.get("PATH_DATASETS", ".")
-AVAIL_GPUS = min(1, torch.cuda.device_count())
-BATCH_SIZE = 256 if AVAIL_GPUS else 64
+BATCH_SIZE = 64
```
Comment (Borda) [inline lightning_examples/basic-gan/gan.py:16]: I may keep it as the final notebook runs on GPU so it gives better results but also 64 is fine for CPU testing

**Review by Borda — COMMENTED** [review]
still WIP or ready to go? :rabbit:

**Comment by Borda** [comment]
the test failer is rather unrelated... #154

File: lightning_examples/basic-gan/gan.py (hunk 2 above)
Comment (rohitgr7) [reply lightning_examples/basic-gan/gan.py:16]: will update!

**Comment by rohitgr7** [comment]
> the test failer is rather unrelated... #154

cool!

**Comment by Borda** [comment]
btw seems this link is dead: https://blogs.nvidia.com/blog/2009/12/16/whats-the-difference-between-a-cpu-and-a-gpu

**Comment by rohitgr7** [comment]
> btw seems this link is dead: https://blogs.nvidia.com/blog/2009/12/16/whats-the-difference-between-a-cpu-and-a-gpu

I can't seem to find it either: https://blogs.nvidia.com/?s=cpu+vs+gpu
nvidia website shows it in their catalog, but it's invalid from their side.

**Comment by otaj** [comment]
> > btw seems this link is dead: https://blogs.nvidia.com/blog/2009/12/16/whats-the-difference-between-a-cpu-and-a-gpu
> 
> I can't seem to find it either: https://blogs.nvidia.com/?s=cpu+vs+gpu nvidia website shows it in their catalog, but it's invalid from their side.

That is weird, for me the link works - https://blogs.nvidia.com/blog/2009/12/16/whats-the-difference-between-a-cpu-and-a-gpu/. Could they have it somewhat filtered by location?

**Comment by rohitgr7** [comment]
> > > btw seems this link is dead: https://blogs.nvidia.com/blog/2009/12/16/whats-the-difference-between-a-cpu-and-a-gpu
> > 
> > 
> > I can't seem to find it either: https://blogs.nvidia.com/?s=cpu+vs+gpu nvidia website shows it in their catalog, but it's invalid from their side.
> 
> That is weird, for me the link works - https://blogs.nvidia.com/blog/2009/12/16/whats-the-difference-between-a-cpu-and-a-gpu/. Could they have it somewhat filtered by location?

WORKING NOW!  HAHA..
WEIRD

good to go now then :)

File: lightning_examples/datamodules/datamodules.py [hunk 3]
```diff
@@ -18,8 +19,8 @@
 from torchvision.datasets import CIFAR10, MNIST
 
 PATH_DATASETS = os.environ.get("PATH_DATASETS", ".")
-AVAIL_GPUS = min(1, torch.cuda.device_count())
-BATCH_SIZE = 256 if AVAIL_GPUS else 64
+BATCH_SIZE = 64
```
Comment (Borda) [inline lightning_examples/datamodules/datamodules.py:22]:
```suggestion
```

File: lightning_examples/cifar10-baseline/baseline.py [hunk 4]
```diff
@@ -21,8 +22,8 @@
 seed_everything(7)
 
 PATH_DATASETS = os.environ.get("PATH_DATASETS", ".")
-AVAIL_GPUS = min(1, torch.cuda.device_count())
-BATCH_SIZE = 256 if AVAIL_GPUS else 64
+BATCH_SIZE = 64
```
Comment (Borda) [inline lightning_examples/cifar10-baseline/baseline.py:25]:
```suggestion
```

File: lightning_examples/basic-gan/gan.py [hunk 5]
```diff
@@ -9,12 +8,13 @@
...
```
Comment (Borda) [inline lightning_examples/basic-gan/gan.py:16]:
```suggestion
```

File: lightning_examples/mnist-hello-world/hello-world.py [hunk 6]
```diff
@@ -11,8 +12,8 @@
 from torchvision.datasets import MNIST
 
 PATH_DATASETS = os.environ.get("PATH_DATASETS", ".")
-AVAIL_GPUS = min(1, torch.cuda.device_count())
-BATCH_SIZE = 256 if AVAIL_GPUS else 64
+BATCH_SIZE = 64
```
Comment (Borda) [inline lightning_examples/mnist-hello-world/hello-world.py:15]:
```suggestion
```

**Review by ethanwharris — APPROVED** [review]

**Comment by rohitgr7** [comment]
so here we are running the scripts as notebooks with multiple devices on GPUs. But no strategy is compatible with interactive environments. Should we run on a single device when running on GPUs (for GPU ci)?

**Comment by Borda** [comment]
> But no strategy is compatible with interactive environments. Should we run on a single device when running on GPUs (for GPU ci)?

yes, lest do it just single GPU :+1:

**Pull request merged**